contrib/storage-ipfs/src/main/java/org/apache/drill/exec/store/ipfs/IPFSScanSpec.java

```diff
@@ -0,0 +1,217 @@
+import com.fasterxml.jackson.annotation.JsonTypeName;
+import io.ipfs.multihash.Multihash;
+import org.apache.drill.common.PlanStringBuilder;
+import org.apache.drill.common.exceptions.UserException;
+import org.apache.drill.exec.store.StoragePluginRegistry;
+import org.apache.drill.shaded.guava.com.google.common.collect.ImmutableSet;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.security.InvalidParameterException;
+import java.util.Set;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+
+@JsonTypeName("IPFSScanSpec")
+public class IPFSScanSpec {
+  private static final Logger logger = LoggerFactory.getLogger(IPFSScanSpec.class);
+
+  public enum Prefix {
+    @JsonProperty("ipfs")
+    IPFS("ipfs"),
+    @JsonProperty("ipns")
+    IPNS("ipns");
+
+    @JsonProperty("prefix")
+    private final String name;
+    Prefix(String prefix) {
+      this.name = prefix;
+    }
+
+    @Override
+    public String toString() {
+      return this.name;
+    }
+
+    @JsonCreator
+    public static Prefix of(String what) {
+      switch (what) {
+        case "ipfs" :
+          return IPFS;
+        case "ipns":
+          return IPNS;
+        default:
+          throw new InvalidParameterException("Unsupported prefix: " + what);
+      }
+    }
+  }
+
+  public enum Format {
+    @JsonProperty("json")
+    JSON("json"),
+    @JsonProperty("csv")
+    CSV("csv");
+
+    @JsonProperty("format")
+    private final String name;
+    Format(String prefix) {
+      this.name = prefix;
+    }
+
+    @Override
+    public String toString() {
+      return this.name;
+    }
+
+    @JsonCreator
+    public static Format of(String what) {
+      switch (what) {
+        case "json" :
+          return JSON;
+        case "csv":
+          return CSV;
+        default:
+          throw new InvalidParameterException("Unsupported format: " + what);
+      }
+    }
+  }
+
+  public static Set<String> formats = ImmutableSet.of("json", "csv");
+  private Prefix prefix;
+  private String path;
+  private Format formatExtension;
+  private final IPFSContext ipfsContext;
+
+  @JsonCreator
+  public IPFSScanSpec (@JacksonInject StoragePluginRegistry registry,
+                       @JsonProperty("IPFSStoragePluginConfig") IPFSStoragePluginConfig ipfsStoragePluginConfig,
+                       @JsonProperty("prefix") Prefix prefix,
+                       @JsonProperty("format") Format format,
+                       @JsonProperty("path") String path) {
+    this.ipfsContext = registry.resolve(ipfsStoragePluginConfig, IPFSStoragePlugin.class).getIPFSContext();
+    this.prefix = prefix;
+    this.formatExtension = format;
+    this.path = path;
+  }
+
+  public IPFSScanSpec (IPFSContext ipfsContext, String path) {
+    this.ipfsContext = ipfsContext;
+    parsePath(path);
+  }
+
+  private void parsePath(String path) {
+    //FIXME: IPFS hashes are actually Base58 encoded, so "0" "O" "I" "l" are not valid
```

Comment:
Again, please either remove, or include a reference to a JIRA to document what needs to be done.

Reply:
Removed in d2ea637.